src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -117,19 +117,7 @@ public void ProcessResult(IDictionary<string, object> functionArguments, object[
                 return;
             }
 
-            HttpRequestMessage request = null;
-            object argValue = null;
-            if (functionArguments.TryGetValue(triggerInputName, out argValue) && argValue is HttpRequestMessage)
-            {
-                request = (HttpRequestMessage)argValue;
-            }
-            else
-            {
-                // No argument is bound to the request message, so we should have 
-                // it in the system arguments
-                request = systemArguments.FirstOrDefault(a => a is HttpRequestMessage) as HttpRequestMessage;
-            }
-
+            HttpRequestMessage request = (HttpRequestMessage)functionArguments.Values.Union(systemArguments).FirstOrDefault(p => p is HttpRequestMessage);
```

Comment:
The raw HttpRequestMessage might be a user defined parameter, or if they didn't define one, it will be a system parameter we injected.